src/track-gpc-signal.ts

```diff
@@ -0,0 +1,21 @@
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { log } from '@guardian/libs';
+import { EventTimer } from './event-timer';
+
+/**
+ * Collect metrics on gpcSignal presence and value
+ * https://globalprivacycontrol.github.io/gpc-spec/
+ */
+const initTrackGpcSignal = (consentState: ConsentState) => {
+	// If undefined we set the property value to -1, false is 0, true is 1
+	const gpcSignal =
+		consentState.gpcSignal === undefined ? -1 : +consentState.gpcSignal;
```

Comment:
Why do we actually need to create this dependency on CMP here? What does the indirection give us?
Can't we just do `navigator.globalPrivacyControl` ?

Reply:
We could, but I prefer keeping the CMP the source of truth. 

For example the GPC can also be signalled by a HTTP header, and we might implement some mechanism to notice that and we'd want to keep the logic for saying the "GPC is ON" in one place, likewise future potential extensions or whatever might lead us down the same path.

Reply:
Fair point and well made.